File: classes/pokemon.js
const natures = [
    "Hardy",
    "Lonely",
    "Brave",
    "Adamant",
    "Naughty",
    "Docile",
    "Bold",
    "Relaxed",
    "Impish",
    "Lax",
    "Serious",
    "Timid",
    "Hasty",
    "Jolly",
    "Naive",
    "Bashful",
    "Modest",
    "Mild",
    "Quiet",
    "Rash",
    "Quirky",
    "Calm",
    "Gentle",
    "Sassy",
    "Careful"
]
module.exports = class Pokemon {
    constructor(object) {
        this.xp = 1;
        this.name = object.name;
        this.price = 0;
        this.id = null;
        this.event = false;
        this.level = object.level;
        this.gender = object.gender;
        this.index = object.index;
        this.url = object.url;
        this.hp =  Math.floor(Math.random() * 31);
		this.atk =  Math.floor(Math.random() * 31);
		this.def =  Math.floor(Math.random() * 31);
		this.spatk =  Math.floor(Math.random() * 31);
		this.spdef = Math.floor(Math.random() * 31);
		this.speed =  Math.floor(Math.random() * 31);
        this.moves = [];
        this.helditem = [];
        this.shiny = object.shiny;
        this.rarity = object.rarity;
        this.nature = natures[Math.floor(Math.random() * natures.length)];
        let totaliv = ((this.hp + this.atk + this.def + this.spatk + this.spdef + this.speed) / 186) * 100;
        this.totalIV = (totaliv.toFixed(1));
    }
}

File: Classes/Pokemon.py
import random

natures = [
    "Hardy", "Lonely", "Brave", "Adamant", "Naughty", "Docile", "Bold", "Relaxed",
    "Impish", "Lax", "Serious", "Timid", "Hasty", "Jolly", "Naive", "Bashful",
    "Modest", "Mild", "Quiet", "Rash", "Quirky", "Calm", "Gentle", "Sassy", "Careful"
]

class Pokemon:
    def __init__(self, object, lvl=1, xp=0):
        self.level = lvl
        self.xp = xp
        self.name = object["name"]
        self.url = object["url"]
        self.hp = random.randint(0, 31)
        self.atk = random.randint(0, 31)
        self.defense = random.randint(0, 31)
        self.spatk = random.randint(0, 31)
        self.spdef = random.randint(0, 31)
        self.speed = random.randint(0, 31)
        self.moves = []
        self.shiny = object["shiny"]
        self.rarity = object["rarity"]
        self.xp = 0
        self.nature = random.choice(natures)
        totaliv = ((self.hp + self.atk + self.defense + self.spatk + self.spdef + self.speed) / 186) * 100
        self.totalIV = round(totaliv, 2)
        # self.fav = False